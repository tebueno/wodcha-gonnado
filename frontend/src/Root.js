import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import reduxPromise from 'redux-promise';
import reducers from 'reducers';
import HomePage from 'pages/Home';
import WodPage from 'pages/Wod';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

export default ({ children }) => {
    const store = createStore(
      reducers,
      applyMiddleware(reduxPromise)
    );
  
    return <Provider store={store}>
    <BrowserRouter>
      <Switch>
        <Route exact path='/' component={HomePage} />
        <Route exact path='/workout/:wodId' component={WodPage} />
      </Switch>
    </BrowserRouter>
    {children}</Provider>;
};