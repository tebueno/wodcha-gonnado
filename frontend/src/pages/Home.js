import React, { Component } from "react";
import Card from 'components/card/card';
import 'css/layout.css';
import { connect } from 'react-redux';
import * as actions from 'actions';

class HomePage extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.page = 1;
    this.props.fetchWods(this.page).then(() => this.addSpanRow());
    window.addEventListener('scroll', this.handleScroll)
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.handleScroll);
  }

  handleScroll = () => {
    let container = -1 * document.querySelector('#masonry').getBoundingClientRect().top + window.innerHeight;
    let height = Math.floor(document.querySelector('#masonry').getBoundingClientRect().height);

    if (container >= height) { 
       this.page++;
       this.props.fetchWods(this.page).then(() => this.addSpanRow())
    }
  };

  addSpanRow = () => {

    // TODO: make this less brittle
    document.querySelectorAll('.card > div > span ')
      .forEach(item => {
        const height = item.offsetHeight;
        const rowSpan = Math.ceil((height + 25) / 125); 
        item.parentElement.parentElement.style.gridRowEnd = `span ${rowSpan}`;
    });

  }

  render() {
    const cards = this.props.wods.map(wod => <Card key={wod.id} {...wod} />);

    return (
      <div id='masonry'>
      {cards}
      </div>
    )
  }
}

const mapStateToProps = (state) => { 
  return { wods: state.wods }
}

export default connect(mapStateToProps, actions)(HomePage);
