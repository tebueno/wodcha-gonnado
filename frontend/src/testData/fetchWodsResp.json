[
  {
    "_id": "5c0747cf9b78ae203b86fa6f",
    "link": "https://www.crossfitinvictus.com/wod/november-14-2018-fitness/",
    "status": "success",
    "workout": "\n\n\nA.\nThree sets of:\nDumbbell Bench Press x 10-12 reps @ 2011\nRest 60 seconds\nStrict Pull-Ups x 10-12 reps @ 2010\nRest 2 minutes\nB.\nEvery 6 minutes, for 18 minutes (3 sets) for times:\nRow 500 Meters\n25 Push-Ups\nRun 200 Meters\n",
    "__v": 0
  },
  {
    "_id": "5c0747d09b78ae203b86fa70",
    "link": "https://www.crossfitinvictus.com/wod/november-13-2018-fitness/",
    "status": "success",
    "workout": "\n\n\nA.\nThree rounds for time of:\n30 Kettlebell Swings\n20 Jumping Lunges\nRest until relatively recovered, and then…\nB.\nThree sets of:\nPosted Single-Leg Deadlifts x 8-10 reps each @ 3011\nRest 60 seconds\nBarbell Glute Bridges x 6-8 reps @ 20X1\nRest 60 seconds\nSide Plank x 45 seconds each side\nRest 60 seconds\n",
    "__v": 0
  },
  {
    "_id": "5c0747d19b78ae203b86fa71",
    "link": "https://www.crossfitinvictus.com/wod/november-12-2018-performance-fitness/",
    "status": "success",
    "workout": "\n\n\nWorkout of the Day\nThe Three Wise Men Tribute\n“Jeremy”\nComplete as many rounds and reps as possible in 4 minutes of:\n5 Hang Squat Snatch\n10 Burpees Over the Barbell\n(Elite = 185/125 lbs, RX = 135/95 lbs, Scaled = 95/65 lbs)\nRest 2 minutes, and then…\n“Ben”\nComplete as many rounds and reps as possible in 4 minutes of:\n10 Power Cleans\n20 Pull-Ups\n(Elite = 185/125 lbs, RX = 135/95 lbs, Scaled = 95/65 lbs)\nRest 2 minutes, and then…\n“Beau”\nComplete as many rounds and reps as possible in 4 minutes of:\n15 Box Jump-Overs\n30 Wall Ball Shots\n(Elite & RX = 24”/20”; Scaled – 24”/20” Step-Overs Allowed)\n(Elite & RX = 20/14 lbs; Scaled – 14/8 lbs)\n————————————\nToday is Veteran’s Day! Please do more than just complete the Three Wise Men Tribute workout…please also honor America’s veterans and make a donation – no matter how small – to funds dedicated improving veterans’ services. There are many different organizations out there that do a fantastic job, so if you have one, please give it your support. If you’re looking for a good organization to support, please join us in our support of HeadStrong and The Honor Foundation.\n",
    "__v": 0
  },
  {
    "_id": "5c0747d29b78ae203b86fa72",
    "link": "https://www.crossfitinvictus.com/wod/november-11-2018/",
    "status": "success",
    "workout": "\n\n\nRecovery Day\nA.\nMobility and Maintenance\n* Choose 1-2 Thoracic Mobility Drills from Kelly Starrett’s Mobility WOD and spend 5-10 minutes with them.\n* Choose 2-3 Lower Body Mobility Drills from Kelly Starrett’s Mobility WOD and spend 10-12 minutes with them.\n* Choose 1-2 Upper Extremity Mobility Drills from Kelly Starrett’s Mobility WOD and spend 5-10 minutes with them.\n– AND/OR –\n* Body work from a licensed body worker (ART, Graston, acupuncture, etc…)\nB.\nNutrition Preparation\n* Ensure that you have quality foods prepared in the appropriate quantities to fuel your efforts for the remainder of the week.\nC.\nMental Restoration\n* Different for everyone – could be meditation or could be gathering with friends; the key is to engage in activities that make your life full and help you recharge your batteries.\n",
    "__v": 0
  },
  {
    "_id": "5c0747d39b78ae203b86fa73",
    "link": "https://www.crossfitinvictus.com/wod/november-10-2018-fitness/",
    "status": "success",
    "workout": "\n\n\nIn teams of two, complete three rounds for time of:\n800 Meter Run (partners must run together)\n80 Kettlebell Swings\n40 Goblet Squats\n20 Bar-Facing Burpees\nPartners may partition the repetitions however they choose, but only one partner may be working at a time – except on the run, which must be completed together.\n",
    "__v": 0
  },
  {
    "_id": "5c0c08d68b513385138d6eea",
    "link": "https://www.crossfitinvictus.com/wod/november-18-2018/",
    "status": "success",
    "workout": "\n\n\nRecovery Day\nA.\nMobility and Maintenance\n* Choose 1-2 Thoracic Mobility Drills from Kelly Starrett’s Mobility WOD and spend 5-10 minutes with them.\n* Choose 2-3 Lower Body Mobility Drills from Kelly Starrett’s Mobility WOD and spend 10-12 minutes with them.\n* Choose 1-2 Upper Extremity Mobility Drills from Kelly Starrett’s Mobility WOD and spend 5-10 minutes with them.\n– AND/OR –\n* Body work from a licensed body worker (ART, Graston, acupuncture, etc…)\nB.\nNutrition Preparation\n* Ensure that you have quality foods prepared in the appropriate quantities to fuel your efforts for the remainder of the week.\nC.\nMental Restoration\n* Different for everyone – could be meditation or could be gathering with friends; the key is to engage in activities that make your life full and help you recharge your batteries.\n",
    "__v": 0
  },
  {
    "_id": "5c0c08d78b513385138d6eeb",
    "link": "https://www.crossfitinvictus.com/wod/november-17-2018-fitness/",
    "status": "success",
    "workout": "\n\n\nIn teams of two, alternating movements throughout, complete as many rounds and reps as possible in 30 minutes of:\n25 Mountain Climbers\n20 Kettlebell Swings\n15 Wall Ball Shots\n10 Alternating Reverse Lunges with Dumbbells\n5 Toes to Bar or Hanging Leg Raises\nPartner A starts with 25 Mountain Climbers, Partner B then performs 20 kettlebell swings, Partner A then performs 15 wall ball shots, Partner B does 10 reverse lunges, Partner A does 5 toes to bar or leg raises, then Partner B does 25 mountain climbers…and so on. \n",
    "__v": 0
  },
  {
    "_id": "5c0c08d98b513385138d6eec",
    "link": "https://www.crossfitinvictus.com/wod/november-16-2018-fitness/",
    "status": "success",
    "workout": "\n\n\nA.\nEvery 2 minutes, for 18 minutes (3 sets of each):\nStation 1 – Back Squat x 8-10 reps @ 30X1\nStation 2 – Supine Ring Rows x 10-12 reps @ 2111\nStation 3 – Medicine Ball Hamstring Curls x 10-12 reps @ 2111\nB.\nFor time:\n800 Meter Run\n80 Alternating Dumbbell Snatches\n80-Foot Walking Lunges with Dumbbell Farmer’s Carry\n800 Meter Run\n",
    "__v": 0
  },
  {
    "_id": "5c0c08da8b513385138d6eed",
    "link": "https://www.crossfitinvictus.com/wod/november-15-2018-fitness/",
    "status": "success",
    "workout": "\n\n\nA.\nEvery 90 seconds, for 18 minutes (3 sets of each):\nStation 1 – Snatch-Grip Romanian Deadlift x 6-8 reps @ 4011\nStation 2 – Alternating Cossack Squats x 12 reps @ 2111\nStation 3 – Supinated-Grip Hang from Pull-Up Bar x 30-45 seconds\nStation 4 – L-Sit Taps x 40 seconds (10 each leg x 2 sets)\nB.\nEvery 90 seconds, for 15 minutes (5 sets of each):\nStation 1 – 12 Alternating Reverse Lunges with Kettlebell Farmer’s Carry\nStation 2 – 18/12 Calories of Assault Bike or Rowing\n",
    "__v": 0
  },
  {
    "_id": "5c0c08db8b513385138d6eee",
    "link": "https://www.crossfitinvictus.com/wod/november-14-2018-fitness/",
    "status": "success",
    "workout": "\n\n\nA.\nThree sets of:\nDumbbell Bench Press x 10-12 reps @ 2011\nRest 60 seconds\nStrict Pull-Ups x 10-12 reps @ 2010\nRest 2 minutes\nB.\nEvery 6 minutes, for 18 minutes (3 sets) for times:\nRow 500 Meters\n25 Push-Ups\nRun 200 Meters\n",
    "__v": 0
  },
  {
    "_id": "5c0c08dd8b513385138d6eef",
    "link": "https://www.crossfitinvictus.com/wod/november-13-2018-fitness/",
    "status": "success",
    "workout": "\n\n\nA.\nThree rounds for time of:\n30 Kettlebell Swings\n20 Jumping Lunges\nRest until relatively recovered, and then…\nB.\nThree sets of:\nPosted Single-Leg Deadlifts x 8-10 reps each @ 3011\nRest 60 seconds\nBarbell Glute Bridges x 6-8 reps @ 20X1\nRest 60 seconds\nSide Plank x 45 seconds each side\nRest 60 seconds\n",
    "__v": 0
  },
  {
    "_id": "5c0c08de8b513385138d6ef0",
    "link": "https://www.crossfitinvictus.com/wod/november-12-2018-performance-fitness/",
    "status": "success",
    "workout": "\n\n\nWorkout of the Day\nThe Three Wise Men Tribute\n“Jeremy”\nComplete as many rounds and reps as possible in 4 minutes of:\n5 Hang Squat Snatch\n10 Burpees Over the Barbell\n(Elite = 185/125 lbs, RX = 135/95 lbs, Scaled = 95/65 lbs)\nRest 2 minutes, and then…\n“Ben”\nComplete as many rounds and reps as possible in 4 minutes of:\n10 Power Cleans\n20 Pull-Ups\n(Elite = 185/125 lbs, RX = 135/95 lbs, Scaled = 95/65 lbs)\nRest 2 minutes, and then…\n“Beau”\nComplete as many rounds and reps as possible in 4 minutes of:\n15 Box Jump-Overs\n30 Wall Ball Shots\n(Elite & RX = 24”/20”; Scaled – 24”/20” Step-Overs Allowed)\n(Elite & RX = 20/14 lbs; Scaled – 14/8 lbs)\n————————————\nToday is Veteran’s Day! Please do more than just complete the Three Wise Men Tribute workout…please also honor America’s veterans and make a donation – no matter how small – to funds dedicated improving veterans’ services. There are many different organizations out there that do a fantastic job, so if you have one, please give it your support. If you’re looking for a good organization to support, please join us in our support of HeadStrong and The Honor Foundation.\n",
    "__v": 0
  },
  {
    "_id": "5c0c08e08b513385138d6ef1",
    "link": "https://www.crossfitinvictus.com/wod/november-11-2018/",
    "status": "success",
    "workout": "\n\n\nRecovery Day\nA.\nMobility and Maintenance\n* Choose 1-2 Thoracic Mobility Drills from Kelly Starrett’s Mobility WOD and spend 5-10 minutes with them.\n* Choose 2-3 Lower Body Mobility Drills from Kelly Starrett’s Mobility WOD and spend 10-12 minutes with them.\n* Choose 1-2 Upper Extremity Mobility Drills from Kelly Starrett’s Mobility WOD and spend 5-10 minutes with them.\n– AND/OR –\n* Body work from a licensed body worker (ART, Graston, acupuncture, etc…)\nB.\nNutrition Preparation\n* Ensure that you have quality foods prepared in the appropriate quantities to fuel your efforts for the remainder of the week.\nC.\nMental Restoration\n* Different for everyone – could be meditation or could be gathering with friends; the key is to engage in activities that make your life full and help you recharge your batteries.\n",
    "__v": 0
  },
  {
    "_id": "5c0c08e18b513385138d6ef2",
    "link": "https://www.crossfitinvictus.com/wod/november-10-2018-fitness/",
    "status": "success",
    "workout": "\n\n\nIn teams of two, complete three rounds for time of:\n800 Meter Run (partners must run together)\n80 Kettlebell Swings\n40 Goblet Squats\n20 Bar-Facing Burpees\nPartners may partition the repetitions however they choose, but only one partner may be working at a time – except on the run, which must be completed together.\n",
    "__v": 0
  },
  {
    "_id": "5c0c08e48b513385138d6ef3",
    "link": "https://www.crossfitinvictus.com/wod/november-9-2018-fitness/",
    "status": "success",
    "workout": "\n\n\nIn teams of two, alternating movements, complete as many rounds and reps as possible in 30 minutes of:\n20 Calories of Assault Bike or Concept 2 Bike Erg\n20 Dumbbell Thrusters\n20 Dumbbell Box Step-Overs\n10 Strict Pull-Ups\n100 Meter Run\nPartners will alternate movements, so Partner A performs 20 Calories of biking, Partner B performs Thrusters, Partner A performs Step-Overs…and so on.\n",
    "__v": 0
  }
]

