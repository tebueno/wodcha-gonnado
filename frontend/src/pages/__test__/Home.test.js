import { mount, shallow } from 'enzyme';
import createRouterContext from 'react-router-test-context';
import toJson from 'enzyme-to-json';
import React from 'react';
import Root from 'Root';
import Home from 'pages/Home';
import testData from 'testData/fetchWodsResp';
import fetchMock from 'fetch-mock';

// TODO: add test for:
// infinity scroll
// clicking workout card
// better validations/assertions for existing case

let wrapped;
const map = {};
window.addEventListener = jest.fn((event, cb) => {
  map[event] = cb;
});
describe('Test rendering of Card component', () => {
  beforeEach(() => {
    const context = createRouterContext();
    wrapped = mount(
      <Root>
        <Home />
      </Root>,
      { context }
    );

    /*     fetchMock.getOnce('http://localhost/api/wods/all?size=15&page=2', {
        "id": "5c0747d29b78ae203b06fa72",
        "wod": "Recovery Day\nA.\nMobility and Maintenance\n* Choose 1-2 Thoracic Mobility Drills from Kelly Starrett’s Mobility WOD and spend 5-10 minutes with them.\n* Choose 2-3 Lower Body Mobility Drills from Kelly Starrett’s Mobility WOD and spend 10-12 minutes with them.\n* Choose 1-2 Upper Extremity Mobility Drills from Kelly Starrett’s Mobility WOD and spend 5-10 minutes with them.\n– AND/OR –\n* Body work from a licensed body worker (ART, Graston, acupuncture, etc…)\nB.\nNutrition Preparation\n* Ensure that you have quality foods prepared in the appropriate quantities to fuel your efforts for the remainder of the week.\nC.\nMental Restoration\n* Different for everyone – could be meditation or could be gathering with friends; the key is to engage in activities that make your life full and help you recharge your batteries.\n"
      });*/
  });

  it('Renders correctly', () => {
    /* map.scroll({height: 111}); */
    /*   Home.contextTypes = {
    router: React.PropTypes.object
  }; */
  fetchMock.getOnce('http://localhost/api/wods/all?size=15&page=1', testData);

    expect(toJson(wrapped)).toMatchSnapshot();
  });

  afterEach(() => {
    wrapped.unmount();
  });

});
