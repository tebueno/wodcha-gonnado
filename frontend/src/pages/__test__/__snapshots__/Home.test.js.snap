// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test rendering of Card component Renders correctly 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <BrowserRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
          >
            <Connect(HomePage)
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <HomePage
                changeWorkoutTxt={[Function]}
                fetchWodById={[Function]}
                fetchWods={[Function]}
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                removeMovement={[Function]}
                searchMovements={[Function]}
                wods={Array []}
              >
                <div
                  id="masonry"
                />
              </HomePage>
            </Connect(HomePage)>
          </Route>
        </Switch>
      </Router>
    </BrowserRouter>
    <Connect(HomePage)>
      <HomePage
        changeWorkoutTxt={[Function]}
        fetchWodById={[Function]}
        fetchWods={[Function]}
        removeMovement={[Function]}
        searchMovements={[Function]}
        wods={Array []}
      >
        <div
          id="masonry"
        />
      </HomePage>
    </Connect(HomePage)>
  </Provider>
</_default>
`;
